smt/expr.h

```diff
@@ -90,6 +89,7 @@ class expr {
   bool isAllOnes() const;
   bool isSMin() const;
   bool isSMax() const;
+  Z3_app isApp() const;
```

Comment:
Same here: this probably shouldn't be exposed either since I don't really want users to rely on us using Z3 since we may want to have multiple backends.
Can you tell me what do you need exactly?

Reply:
(Combined answer for both)
I need a list of expressions corresponding to input variables matching some criteria.  
(i.e. two separate lists corresponding to variables and symbolic constants)

I am trying to extract this from StateValue instances which look like:  
`(ite (= ty_%var_a #b00) %var_a undef_1)`  

Is there another way to get the SMT expression for %var_a?

Reply:
Well, if you just want the inputs, you can ask the Function for the inputs. State also stores quite a bit of info.
You probably don't need to traverse the formula, That's very brittle anyway since encodings can (and will) change over time. We can expose further APIs if information is missing.  I just need to know exactly what's missing/needed :)

Reply:
Using the function inputs seems to require traversing formulas too, at this moment.  
```
  for (auto &i : t.tgt.getInputs()) {
    std::cerr << i.toSMT(tgt_state) << std::endl;
  }
```
The SMT expressions I get from this look similar to the what I had before:
`(ite (= ty_%var_x #b00) %var_x undef_11) / (= #b0 ((_ extract 1 1) ty_%var_x))
`

Can we have an API that gives me the smt expr for %var_x (without requiring simplifications)?